### PR TITLE
ES|QL Add topn support for Floats

### PR DESCRIPTION
**Pull request by ChrisHegarty**
This commit adds topn support for Float to the compute engine.

relates https://github.com/elastic/elasticsearch/issues/109178

### CODE REVIEW

**Comment by ChrisHegarty** [comment]
@elasticmachine update branch

File: x-pack/plugin/esql/compute/src/test/java/org/elasticsearch/compute/operator/topn/TopNOperatorTests.java [hunk 1]
```diff
@@ -505,7 +520,7 @@ public void testCollectAllValues() {
         encoders.add(DEFAULT_SORTABLE);
 
         for (ElementType e : ElementType.values()) {
-            if (e == ElementType.UNKNOWN || e == COMPOSITE || e == FLOAT) {
+            if (e == ElementType.UNKNOWN || e == COMPOSITE) {
```
Comment (ChrisHegarty) [inline x-pack/plugin/esql/compute/src/test/java/org/elasticsearch/compute/operator/topn/TopNOperatorTests.java:523]: I added this exclusions for FLOAT when adding FloatBlocks, the exclusions can now be removed.

**Comment by ChrisHegarty** [comment]
@elasticmachine update branch

**Review by dnhatn — APPROVED** [review]
LGTM.

**Comment by ChrisHegarty** [comment]
This commit adds topn support for Float to the compute engine.

**Pull request merged**